File: Wirtualny swiat-java/Animals/Turtle.java
package Animals;

import java.util.Random;

import World.Organism;
import World.Animal;
import World.GameWorld;
import World.Point;

public class Turtle extends Animal {
    private final Random random = new Random();

    public Turtle(int x, int y, GameWorld world) {

        super(2, 1, x, y, world, 'T');
    }

    @Override
    public void action() {
        int changePosition = random.nextInt(4);
        if (changePosition == 1) {
            super.action(); // tylko w 25% przypadków wykonuje ruch
        }
    }

    @Override
    public void draw() {
        System.out.print(symbol);
    }

    @Override
    public boolean isBlockAttack(Organism attacker) {
        if (attacker.getStrength() < 5) {
            return true;
        } else {
            return false;
        }
    }
}

File: Wirtualny Swiat C++/animals/Turtle.cpp
//
// Created by Dorota on 17.04.2025.
//

#include "Turtle.h"

#include <iostream>
using namespace std;

Turtle::Turtle(int x, int y, GameWorld *world): Animal(2, 1, x, y, world,'T') {

}

void Turtle::action() {
    int changePosition = rand()%4;
    if (changePosition==1) {
        Animal::action(); //dziedziczenie ruchu
    }
}

void Turtle::draw() {
    cout<<symbol;
}

bool Turtle::isBlockAttack(Organism *attacker) {
    if (attacker->GetStrength()<5) {
        return true;
    }
    return false;
}



